viewing_party/party.py

```diff
@@ -1,23 +1,188 @@
 # ------------- WAVE 1 --------------------
+# test_wave_01 - line 7 to line 59
+from logging import StringTemplateStyle
 
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating:
+        return None
+    new_movie = {'title':title,
+    'genre': genre,
+    'rating': rating}
+    return new_movie
 
-# -----------------------------------------
-# ------------- WAVE 2 --------------------
-# -----------------------------------------
+# test_wave_01 - line 61 to line 81
+def add_to_watched(user_data,movie):
+    user_data['watched'].append(movie)
+    return user_data
 
+# test_wave_01 line 104 to line 121  
+def add_to_watchlist(user_data, movie):
+    user_data['watchlist'].append(movie)
+    return user_data
+# test_wave_01
+def watch_movie(user_data,title):
+    
+    watch_list = user_data['watchlist']
+    for index in range(len(watch_list)):
+        movie = watch_list[index]
+        if movie['title']==title:
+            watch_list.pop(index)
+            user_data['watched'].append(movie)
+    return user_data
 
-# -----------------------------------------
-# ------------- WAVE 3 --------------------
-# -----------------------------------------
+# -----------test_wave_02----------
+# line 5 to line 28 on test_wave02
+
+# user_data is a dict with they key=watch list with a dict of movies. 
+# movie has 3 keys: title, genre, rating
+    # movie_list=user_data['watched']
+def get_watched_avg_rating(user_data):
+    print(user_data)
+    count=0
+    score_sum=0
+    
+    if user_data['watched']==[]:
+        average_rating=0.0
+    else:
+        for movie in user_data['watched']:
+            score_sum+= movie['rating']
+            count+=1
+        average_rating=score_sum/count
+    return average_rating
+
+# testwave 02 - line 29 on   
+def get_most_watched_genre(user_data):
+    movie_list=user_data['watched']
+    genre_list=[]
+    genre_dict={}
+    count=0
+    if movie_list==[]:
+        return None
+    else:
+        for movie in movie_list:
+            genre_list.append(movie['genre'])
+        print(f' the genre list is {genre_list}')            
+    for i in genre_list:
+        if i in genre_dict:
+            genre_dict[i]+=1
+        else:
+            genre_dict[i]=1

```

Comment:
I think we can combine this into one for loop:

```suggestion
    for movie in movie_list:
        if movie["genre"] in genre_dict:
            genre_dict[movie["genre"]]+=1
        else:
            genre_dict[movie["genre"]]=1
```

Now we don't need `genre_list` variable at all!